docker/run-one.sh

```diff
@@ -37,7 +37,7 @@ else
 
   echo "Started $GITHUB_USER/$BRANCH on $BUILD at $(date)"
 
-  docker run $VOLUME "$TAG" /bin/bash -c \
+  docker run $VOLUME "$TAG" /bin/bash --login -c \
```

Comment:
This affects all images, right? So, there is no side-effect on other OSes?

Reply:
Yes, it does affect all images. It now uses a login-shell to start the container. CMIW, I don't think it will break any image since all docker images that we have are pretty simple.

Reply:
https://stackoverflow.com/questions/57320549/do-dockerfile-run-commands-run-in-a-login-shell-environment

Reply:
If I tried to CentOS on this PR, it seems that it has side-effects, @wgtmac . Did you try to run all the other OSes actually?
```
$ ./run-one.sh local main centos7
Started local run for ORC-1320 on centos7 at Mon Nov 28 18:17:02 PST 2022
/etc/profile.d/lang.sh: line 19: warning: setlocale: LC_CTYPE: cannot change locale (en_US.UTF-8): No such file or directory
/etc/profile.d/lang.sh: line 20: warning: setlocale: LC_COLLATE: cannot change locale (en_US.UTF-8): No such file or directory
/etc/profile.d/lang.sh: line 23: warning: setlocale: LC_MESSAGES: cannot change locale (en_US.UTF-8): No such file or directory
/etc/profile.d/lang.sh: line 26: warning: setlocale: LC_NUMERIC: cannot change locale (en_US.UTF-8): No such file or directory
/etc/profile.d/lang.sh: line 29: warning: setlocale: LC_TIME: cannot change locale (en_US.UTF-8): No such file or directory
-- The C compiler identification is GNU 9.3.1
-- The CXX compiler identification is GNU 9.3.1
...
```

Reply:
> If I tried to use this PR to run CentOS test, it seems that it has side-effects, @wgtmac . Did you try to run all the other OSes actually?
> 
> ```
> $ ./run-one.sh local main centos7
> Started local run for ORC-1320 on centos7 at Mon Nov 28 18:17:02 PST 2022
> /etc/profile.d/lang.sh: line 19: warning: setlocale: LC_CTYPE: cannot change locale (en_US.UTF-8): No such file or directory
> /etc/profile.d/lang.sh: line 20: warning: setlocale: LC_COLLATE: cannot change locale (en_US.UTF-8): No such file or directory
> /etc/profile.d/lang.sh: line 23: warning: setlocale: LC_MESSAGES: cannot change locale (en_US.UTF-8): No such file or directory
> /etc/profile.d/lang.sh: line 26: warning: setlocale: LC_NUMERIC: cannot change locale (en_US.UTF-8): No such file or directory
> /etc/profile.d/lang.sh: line 29: warning: setlocale: LC_TIME: cannot change locale (en_US.UTF-8): No such file or directory
> -- The C compiler identification is GNU 9.3.1
> -- The CXX compiler identification is GNU 9.3.1
> ...
> ```

I have tried some of them. The warning you have seen above comes from `source /etc/bashrc`